### PR TITLE
AESinkFactory: Adjust Sndio

### PR DESCRIPTION
**Pull request by ghost**
## Description

Adjust Sndio backend as requested by @fritsch in #13119.

If you need someone to adjust OSS as well, please let me know.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
Not tested myself, but I think you need something like #13181 too, as not all files include `system.h` (where `HAS_SNDIO` is defined).

File: xbmc/cores/AudioEngine/Sinks/AESinkSNDIO.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "system.h"
 #ifdef HAS_SNDIO
 #include "AESinkSNDIO.h"
+#include "cores/AudioEngine/Engines/ActiveAE/ActiveAE.h"
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Sinks/AESinkSNDIO.cpp]: if this is required, something is wrong with this sink

Comment (t6) [reply xbmc/cores/AudioEngine/Sinks/AESinkSNDIO.cpp]: Addressed both review items. Anything else?

**Comment by Rechi** [comment]
jenkins build this please

File: xbmc/windowing/X11/WinSystemX11GLContext.cpp [hunk 2]
```diff
@@ -63,11 +63,18 @@ CWinSystemX11GLContext::CWinSystemX11GLContext()
...
+  else if (StringUtils::EqualsNoCase(envSink, "SNDIO"))
+  {
+    X11::SndioRegister();
+  }
   else
   {
     if (!X11::PulseAudioRegister())
     {
-      X11::ALSARegister();
+      if (!X11::ALSARegister())
```
Comment (fritsch) [inline xbmc/windowing/X11/WinSystemX11GLContext.cpp]: ALSA cannot return false for now - it's hardcoded to return true

Comment (fritsch) [reply xbmc/windowing/X11/WinSystemX11GLContext.cpp]: Minus: It is not compiled in :-)

**Comment by fritsch** [comment]
Jenkins is fine (minus the float issue). I will push that as my PulseRework needs rebasing ontop of this.

Thanks very much @t6 for reacing that fast.

File: xbmc/windowing/X11/WinSystemX11GLContext.cpp (hunk 2 above)
Comment (fritsch) [reply xbmc/windowing/X11/WinSystemX11GLContext.cpp]: Which is the case on FreeBSD I assume -> all fine.

**Pull request merged**